ducktape/tests/loader.py

```diff
@@ -80,7 +86,15 @@ def load(self, test_discovery_symbols):
                 raise LoaderException("Didn't find any tests for symbol %s." % symbol)
 
         self.logger.debug("Discovered these tests: " + str(all_test_context_list))
-        return all_test_context_list * self.repeat
+
+        # Select the subset of tests. Select every nth test instead of blocks of n to avoid groups of tests that are
+        # parametrizations of the same test being grouped together since that can lead to a single, parameterized,
+        # long-running test causing a very imbalanced workload across different subsets. Imbalance is still possible,
+        # but much less likely using this heuristic.
+        subset_test_context_list = list(itertools.islice(all_test_context_list, self.subset, None, self.subsets))
```

Comment:
How do we know that the order of tests in the list will remain same ? The =os.walk()= call provides no such guarantees. 
https://docs.python.org/2/library/os.html#os.walk
https://github.com/confluentinc/ducktape/blob/master/ducktape/tests/loader.py#L287
Probably we need to do sort to ensure this.